sql/mozfun/mobile_search/normalized_app_name/udf.sql

```diff
@@ -0,0 +1,57 @@
+
+
+CREATE OR REPLACE FUNCTION `normalized_app_name`(app_name STRING, os STRING)
+RETURNS STRUCT<normalized_app_name STRING, normalized_app_name_os STRING>
+AS
+(
+  STRUCT(
+    CASE
+      WHEN app_name = 'Fenix' AND os = 'Android' THEN 'Firefox' 
+      WHEN app_name = 'Fennec' AND os = 'Other' THEN 'Fennec' 
+      WHEN app_name = 'Fennec' AND os = 'Android' THEN 'Fennec'
+      WHEN app_name = 'Fennec' AND os = 'iOS' THEN 'Firefox'
+      WHEN app_name = 'Firefox Preview' AND os = 'Android' THEN 'Firefox Preview'
+      WHEN app_name = 'FirefoxConnect' AND os = 'Android' THEN 'Firefox for Echo Show'
+      WHEN app_name = 'FirefoxForFireTV' AND os = 'Android' THEN 'Firefox for FireTV'
+      WHEN app_name = 'Focus Android Glean' AND os = 'Android' THEN 'Focus'
+      WHEN app_name = 'Focus iOS Glean' AND os = 'iOS' THEN 'Focus'
+      WHEN app_name = 'Klar Android Glean' AND os = 'Android' THEN 'Klar'
+      WHEN app_name = 'Klar iOS Glean' AND os = 'iOS' THEN 'Klar'
+      WHEN app_name = 'Other' AND os = 'iOS' THEN 'Other'
+      WHEN app_name = 'Other' AND os = 'Other' THEN 'Other'
+      WHEN app_name = 'Other' AND os = 'Android' THEN 'Other'
+      WHEN app_name = 'Zerda' AND os = 'Android' THEN 'Firefox Lite'
+      WHEN app_name = 'Zerda_cn' AND os = 'Android' THEN 'Firefox Lite (China)'
+      ELSE 'Other'
```

Comment:
I wonder if the fallback should be NULL instead? To me the most important thing is to avoid silent failures. I think since `'Other'` is a valid app name, NULL is a better way to indicate a weird value, but what do you think?